web/src/lib/domain-logic/allowFormation.ts

```diff
@@ -1,8 +1,29 @@
-import { PublicFilingLegalType, publicFilingLegalTypes } from "@businessnjgovnavigator/shared";
+import {
+  BusinessPersona,
+  castPublicFilingLegalTypeToFormationType,
+  FormationLegalType,
+  PublicFilingLegalType,
+  publicFilingLegalTypes,
+} from "@businessnjgovnavigator/shared";
+
+export const allowFormation = (
+  legalStructureId: string | undefined,
+  persona: BusinessPersona | undefined
+): boolean => {
+  const featureFlagMap: Partial<Record<FormationLegalType, boolean>> = {
+    "foreign-limited-partnership": process.env.FEATURE_BUSINESS_FLP === "true",
+  };
 
-export const allowFormation = (legalStructureId: string | undefined): boolean => {
   if (publicFilingLegalTypes.includes(legalStructureId as PublicFilingLegalType)) {
-    return true;
+    if (persona === "FOREIGN" || legalStructureId === "nonprofit") {
+      return (
+        featureFlagMap[
+          castPublicFilingLegalTypeToFormationType(legalStructureId as PublicFilingLegalType, persona)
+        ] ?? true
+      );
+    } else {
+      return true;
+    }
```

Comment:
I know this is how it was before, but given the `?? true` on line 22, is the larger if-statement on line 18, and its `else true` necessary?